data/src/client.rs

```diff
@@ -461,6 +469,11 @@ impl Client {
                 }
             }
             Command::JOIN(channel, _) => {
+                if self.supports_away_notify {
+                    // Sends WHO to get away state on users.
+                    let _ = self.sender.try_send(command!("WHO", channel));
+                }
```

Comment:
I noticed we're sending WHO every-time someone joins a channel. We should ensure this is only sent when we're the user: `user.nickname() == self.nickname()`

Reply:
Ah yes, good call.